File: scratch/parse_spotify_rss.py
import urllib.request
import xml.etree.ElementTree as ET
import os

url = "https://anchor.fm/s/114a64400/podcast/rss"
headers = {'User-Agent': 'Mozilla/5.0'}

print("Fetching RSS feed...")
req = urllib.request.Request(url, headers=headers)
try:
    with urllib.request.urlopen(req) as response:
        xml_data = response.read()
    print("RSS Feed fetched successfully. Parsing...")
    
    root = ET.fromstring(xml_data)
    items = root.findall('.//item')
    print(f"Found {len(items)} episodes in feed.\n")
    
    for idx, item in enumerate(items):
        title_elem = item.find('title')
        title = title_elem.text if title_elem is not None else "No Title"
        
        enclosure = item.find('enclosure')
        enc_url = enclosure.get('url') if enclosure is not None else "No URL"
        
        print(f"[{idx+1}] Title: {title}")
        print(f"    URL: {enc_url}")
        print("-" * 50)
        
except Exception as e:
    print(f"Error: {e}")

File: scratch/parse_spotify_rss.dart
import 'dart:io';
import 'dart:convert';
import 'package:xml/xml.dart'; // We can parse via simple string matching/xml if we don't have xml package.
// Let's write a parser using RegExp to avoid external package dependency, ensuring it runs out of the box with standard dart.

void main() async {
  final url = 'https://anchor.fm/s/114a64400/podcast/rss';
  print('Fetching RSS feed from $url...');
  try {
    final client = HttpClient();
    final request = await client.getUrl(Uri.parse(url));
    request.headers.add('User-Agent', 'Mozilla/5.0');
    final response = await request.close();
    if (response.statusCode != 200) {
      print('Failed to load RSS feed: ${response.statusCode}');
      return;
    }
    
    final xmlContent = await response.transform(utf8.decoder).join();
    print('RSS feed fetched. Parsing episodes...');
    
    // Simple item splitting
    final itemRegex = RegExp(r'<item>(.*?)</item>', dotAll: true);
    final titleRegex = RegExp(r'<title><!\[CDATA\[(.*?)\]\]></title>', dotAll: true);
    final titleFallbackRegex = RegExp(r'<title>(.*?)</title>', dotAll: true);
    final enclosureRegex = RegExp(r'<enclosure\s+url="([^"]+)"', dotAll: true);
    
    final matches = itemRegex.allMatches(xmlContent).toList();
    print('Found ${matches.length} episodes.\n');
    
    for (var i = 0; i < matches.length; i++) {
      final itemContent = matches[i].group(1) ?? '';
      
      var title = '';
      final titleMatch = titleRegex.firstMatch(itemContent);
      if (titleMatch != null) {
        title = titleMatch.group(1) ?? '';
      } else {
        final titleFallback = titleFallbackRegex.firstMatch(itemContent);
        if (titleFallback != null) {
          title = titleFallback.group(1) ?? '';
        }
      }
      
      var encUrl = '';
      final encMatch = enclosureRegex.firstMatch(itemContent);
      if (encMatch != null) {
        encUrl = encMatch.group(1) ?? '';
      }
      
      print('[${i + 1}] Title: ${title.trim()}');
      print('    URL: $encUrl');
      print('-' * 60);
    }
  } catch (e) {
    print('Error: $e');
  }
}
